lib/ansible/inventory/manager.py

```diff
@@ -619,9 +619,8 @@ def subset(self, subset_pattern):
             # allow Unix style @filename data
             for x in subset_patterns:
                 if x[0] == "@":
-                    fd = open(x[1:])
-                    results.extend([to_text(l.strip()) for l in fd.read().split("\n")])
-                    fd.close()
+                    with open(x[1:]) as fd:
+                        results.extend([to_text(l.strip()) for l in fd.readlines()])
```

Comment:
This change changes semantics: `fd.read().split('\n')` does not return the newlines, while `fd.readlines()` does.

I would simply do
```suggestion
                        results.extend([to_text(l.strip()) for l in fd.read().split("\n")])
```
instead.